src/extension/prompts/node/agent/fileLinkificationInstructions.tsx

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { PromptElement } from '@vscode/prompt-tsx';
+import { Tag } from '../base/tag';
+
+export class FileLinkificationInstructions extends PromptElement<{}> {
+	render() {
+		return <Tag name='file_linkification'>
+			ALWAYS convert file paths to markdown links with 1-based line numbers whenever you cite specific code locations. Use links inside normal sentences, not as the entire answer.<br />
+			Format examples:<br />
+			- `The handler lives in [path/to/file.ts](path/to/file.ts#L10).` (single line)<br />
+			- `See [path/to/file.ts](path/to/file.ts#L10-L12) for the range.`<br />
+			- `Configuration is defined in [path/to/file.ts](path/to/file.ts).` (whole file)<br />
+			- `The widget renderer attaches anchors ([src/renderer.ts](src/renderer.ts#L42-L48)).` (in parentheses)<br />
+			When you need a bullet list of references with line numbers, you can use descriptive text:<br />
+			- [Await chat view](path/to/chatQuick.ts#L142)<br />
+			- [Show widget](path/to/chatQuick.ts#L321)<br />
+			NEVER cite file paths as plain text when referring to specific locations. For example, instead of saying `The function is in exampleScript.ts at line 25.`, say `The function is in [exampleScript.ts](exampleScript.ts#L25).`<br />
+			Critical rules:<br />
```

Comment:
I don't think this makes much of an impact with most models if they are not included as reminders closer to the user message

Reply:
In my testing, I have found this prompt to make a difference. Without this, line # always appear separate from the file path reference, for e.g. `line 25 in file.ts `or `file.ts at line 25.`